src/net/ossl.cc

```diff
@@ -1202,15 +1202,17 @@ class session : public enable_shared_from_this<session>, public session_impl {
         return ssl_ctx;
     }
 
-    static std::optional<sstring> get_ossl_string(X509_NAME* name){
-        if (auto name_str = X509_NAME_oneline(name, nullptr, 0)) {
-            // sstring constructor may throw, to ensure deallocation of this OpenSSL string in
-            // all cases, wrap the call to free() in a deferred_action
-            auto done = defer([&name_str]() noexcept { OPENSSL_free(name_str); });
-            sstring ossl_str(name_str);
-            return ossl_str;
+    static std::optional<sstring> get_dn_string(X509_NAME* name) {
+        auto out = bio_ptr(BIO_new(BIO_s_mem()));
+        if (!X509_NAME_print_ex(out.get(), name, 0, ASN1_STRFLGS_RFC2253 | XN_FLAG_SEP_COMMA_PLUS |
```

Comment:
> X509_NAME_print_ex() and X509_NAME_print_ex_fp() return 1 on success or 0 on error if the XN_FLAG_COMPAT is set, which is the same as X509_NAME_print(). Otherwise, it returns -1 on error or other values on success.

Should the check be `-1 == X509_NAME_print_ex`?